src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -599,26 +577,28 @@ bool ColorControlServer::shouldExecuteIfOff(EndpointId endpoint, uint8_t optionM
  */
 void ColorControlServer::handleModeSwitch(EndpointId endpoint, uint8_t newColorMode)
 {
-    uint8_t oldColorMode = 0;
+    auto oldColorMode = ColorMode::kCurrentHueAndCurrentSaturation;
     Attributes::ColorMode::Get(endpoint, &oldColorMode);
 
     uint8_t colorModeTransition;
 
-    if (oldColorMode == newColorMode)
+    ColorControl::ColorMode newColorModeEnum                = static_cast<ColorControl::ColorMode>(newColorMode);
+    ColorControl::EnhancedColorMode newEnhanceColorModeEnum = static_cast<ColorControl::EnhancedColorMode>(newColorMode);
+    if (oldColorMode == newColorModeEnum)
     {
         return;
     }
 
-    Attributes::EnhancedColorMode::Set(endpoint, newColorMode);
-    if (newColorMode == ColorControl::EnhancedColorMode::kEnhancedCurrentHueAndCurrentSaturation)
+    Attributes::EnhancedColorMode::Set(endpoint, newEnhanceColorModeEnum);
+    if (newEnhanceColorModeEnum == ColorControl::EnhancedColorMode::kEnhancedCurrentHueAndCurrentSaturation)
     {
         // Transpose COLOR_MODE_EHSV to ColorControl::EnhancedColorMode::kCurrentHueAndCurrentSaturation after setting
         // EnhancedColorMode
-        newColorMode = ColorControl::EnhancedColorMode::kCurrentHueAndCurrentSaturation;
+        newEnhanceColorModeEnum = ColorControl::EnhancedColorMode::kCurrentHueAndCurrentSaturation;
     }
-    Attributes::ColorMode::Set(endpoint, newColorMode);
+    Attributes::ColorMode::Set(endpoint, newColorModeEnum);
```

Comment:
This makes no sense, sorry.  Isn't it potentially setting an invalid value?

I think my previous comment on this function was mis-implemented.  Maybe it's fine if the input is `EnhancedColorModeEnum`, if that's clearly documented and the callers handle it properly.  Certainly having it be uint8_t but then directly casting to `EnhancedColorModeEnum` is no better than passing in `EnhancedColorModeEnum` to start with.

But the mapping from that to `ColorModeEnum` in this function needs to be handled correctly.  And the callers need to map from non-enhanced to enhanced color mode correctly as needed.